datasetinsights/estimators/faster_rcnn.py

```diff
@@ -80,57 +92,111 @@ def __init__(
             box_score_thresh=box_score_thresh,
         )
         self.model_without_ddp = self.model
-        self.gpu = gpu
-        self.rank = rank
         self.sync_metrics = config.get("synchronize_metrics", True)
-        logger.info(f"gpu: {self.gpu}, rank: {self.rank}")
-
-        self.checkpointer = checkpointer
-        checkpoint_file = config.get("checkpoint_file", const.NULL_STRING)
-        if checkpoint_file != const.NULL_STRING:
-            checkpointer.load(self, config.checkpoint_file)
-
         self.metrics = {}
         for metric_key, metric in config.metrics.items():
             self.metrics[metric_key] = EvaluationMetric.create(metric.name)
         self.model.to(self.device)
 
-        if self.config.system.distributed:
+        if self.distributed:
             self.model = torch.nn.parallel.DistributedDataParallel(
                 self.model, device_ids=[self.gpu]
             )
             self.model_without_ddp = self.model.module
 
-    def train(self, **kwargs):
-        """start training, save trained model per epoch."""
+        self.checkpoint_file = checkpoint_file
```

Comment:
Should we have `self.ckeckpoint_file` to be an instance variable of `FasterRCNN` class?

Reply:
Do you mean a variable instead of a instance varable?

Reply:
Yes. Is `self.ckeckpoint_file` used in other places except init? My guess is not.